rust-app/zephyr/src/uart.rs

```diff
@@ -46,10 +52,95 @@ macro_rules! trait_impl {
                     zephyr_sys::syscalls::$context::uart_err_check(core::mem::transmute(device))
                 })
             }
+
+            #[cfg(feature = "have_std")]
+            #[inline(always)]
+            fn uart_config_get(device: &Device) -> Result<UartConfig, i32> {
+                let mut config = UartConfig::default();
+                let rc = unsafe {
+                    zephyr_sys::syscalls::user::uart_config_get(
+                        core::mem::transmute(device),
+                        &mut config.0,
+                    )
+                };
+                if rc == 0 {
+                    Ok(config)
+                } else {
+                    Err(rc)
+                }
```

Comment:
I've been using the `NegErr` trait for this to convert a negative i32 into Err(u32). Since the defines for errnos are positive, bindgen makes them u32. So when testing for a specific error it's a pain to have to covert the u32 const to i32 and negate it.

So you could do
```
unsafe {
    ...
}.neg_err().map(|_| config)